sklearn/linear_model/_omp.py

```diff
@@ -107,13 +107,13 @@ def _cholesky_omp(X, y, n_nonzero_coefs, tol=None, copy_X=True,
                                     overwrite_b=True,
                                     check_finite=False)
             v = nrm2(L[n_active, :n_active]) ** 2
-            Lkk = linalg.norm(X[:, lam]) ** 2 - v
+            Lkk = linalg.norm(X[:, lam], check_finite=False) ** 2 - v
```

Comment:
I think we can do this, because the caller site:

https://github.com/scikit-learn/scikit-learn/blob/24d58668da03239cb1a45a1bd713d5485478a675/sklearn/linear_model/_omp.py#L418-L420

checks that `X` is finite:

https://github.com/scikit-learn/scikit-learn/blob/24d58668da03239cb1a45a1bd713d5485478a675/sklearn/linear_model/_omp.py#L371

But I prefer to not have `check_finite=False` here because it is not obvious that `X` is finite. One needs to go through the code to check that the caller did the check.